Services/CO.CDP.Organisation.WebApi/UseCase/AssignIdentifierUseCase.cs

```diff
@@ -47,35 +62,39 @@ private OrganisationInformation.Persistence.Organisation AssignIdentifier(
         });
 
         return organisation;
-    }
+    }  
 
-    private static void ResetIdentifierPrimaryToFalse(OrganisationInformation.Persistence.Organisation.Identifier? identifier)
+    private static void ResetPrimaryIdentifiers(IEnumerable<OrganisationInformation.Persistence.Organisation.Identifier> identifiers)
     {
-        if (identifier != null)
+        foreach (var identifier in identifiers.Where(id => id.Primary))
         {
             identifier.Primary = false;
-        }    
+        }
     }
 
     public static bool IsPrimaryIdentifier(OrganisationInformation.Persistence.Organisation organisation, string newIdentifierSchemeName)
     {
-        if (newIdentifierSchemeName == IdentifierSchemes.Vat)
+        if (newIdentifierSchemeName == IdentifierSchemes.Vat || !IdentifierSchemesUK.ContainsKey(newIdentifierSchemeName))
         {
             return false;
         }
+       
+        if (organisation.Identifiers.Count == 0)
+        {
+            return true;
+        }
 
-        bool isPrimary = organisation.Identifiers.Count == 0;
+        var primaryIdentifiers = organisation.Identifiers
+           .Where(i => i.Primary && (i.Scheme == IdentifierSchemes.Other || i.Scheme == IdentifierSchemes.Ppon || !IdentifierSchemesUK.ContainsKey(i.Scheme)))
+           .ToList();
 
-        if (organisation.Identifiers.Any(i => i.Scheme == IdentifierSchemes.Other && i.Primary) ||
-            organisation.Identifiers.Any(i => i.Scheme == IdentifierSchemes.Ppon && i.Primary))
+        if (primaryIdentifiers.Any())
         {
-            ResetIdentifierPrimaryToFalse(organisation.Identifiers.FirstOrDefault(i => i.Scheme == IdentifierSchemes.Other && i.Primary));
-            ResetIdentifierPrimaryToFalse(organisation.Identifiers.FirstOrDefault(i => i.Scheme == IdentifierSchemes.Ppon && i.Primary));
-
-            isPrimary = true;
+            ResetPrimaryIdentifiers(primaryIdentifiers);
```

Comment:
I'd feel better if we didn't reset all the identifiers, as that potentially reshuffles them. Resetting all the identifiers means that, If two identifiers can be made primary, the primary identifier might potentially change over time. I think it's safer for integration if we kept the identifier selected as primary first as long as it's assigned to an organisation. 

If there was a primary identifier before, and it's not PPON, it should remain primary.
If there was a primary identifier before, and it is PPON, then:
* if the new identifier can be primary, it should be made primary and PPON should be made secondary
* if the new identifier cannot be made primary, PPON should remain primary

Reply:
@jakzal we are not resetting all the identifiers, it is just Ppon, other and international Identifiers.
May be naming convention made it confusing.